### ISSUE TITLE
why not use the latest edition of node,nwjs,cordova 

### ISSUE THREAD

**Issue by PeterDu0404** [byteball/obyte-gui-wallet#371]

**Comment by tarmo888** [comment]
Because some stuff that was done is Node 6 one way might have been changed to do differently in Node 8 or 10, so there might be changes that need to be done in the code to migrate to new version. Problem is that some of these changes that need to be done might not be in your own code, but in one of the modules that was used, so hopefully there is module is also updated to work on newer Node, but if it isn't then you need to find a new module or modules that does the same thing and replace them, sometimes you can't because you would need to replace a lot of modules or rewrite most of the code.

It is low priority if the Node that you are still using has LTS (it still gets maintenance updates) and there are more important tasks that need to be done first. Updating to newer Node will be invisible change for actual users.

Also, not all developers out there use `nvm` so they might still be on Node 6 or 8 and updating everything to the latest will exclude those developers from participating.

So, having everything on latest version is lot of work and not always possible depending on what other modules are used, more importantly it doesn't affect user experience because users don't care what version of the framework or tool was used to make that executable file.

But great thing about open-source is that, if you have time and patience to update everything to newer versions and you have tested everything thoroughly that all components still work as expected then you can make a Pull Request that fixes this issue.

**Comment by nothingismagick** [comment]
@tarmo888 - I created another issue just about updating the recommendation for node:
#382 

I understand your argument (and the nvm approach), but I still find it confusing and would like to know what parts of this library are not compliant with modern versions of node. You say “might” need changes... what exactly are you talking about? 

Furthermore, the point about excluding developers because of their preferred node version is just blatantly wrong. I don’t care which version I need to use, as long as it is an up-to-date LTS or the latest stable development release. Where I do have a problem, however, is in using outdated and unpatched libraries...

To be clear: my point in the issue I raised #382 is that node 5.12 is not receiving backports for either security or crypto updates. This is a critical issue that needs to be addressed before this Byteball package I can consider it safe to use.

**Comment by tarmo888** [comment]
And my point is that if you would have tried to actually installed it, you would instantly seen the problems that need to be fixed to have it running on higher versions and you would not be asking questions like "Why is node 5.12.0 still recommended?". And maybe even making a PR to fix it.

It is recommended because it is easiest to get it running on that, but you can get it running with node 6 or 8 too, you don't have to be using node 5, it just needs more work on newer. You would know it too, if you would just tried.

**Comment by tarmo888** [comment]
Feel free to go through all of those and make pull requests and make sure it all still works. Keyword there is "new projects". Nobody says that you should stop using it immediately.

**Comment by tonyofbyteball** [comment]
4.0.0 uses Electron instead of nwjs and a lot of dependencies have been upgraded.
